Source/Events.Processing/Filters/FiltersService.cs

```diff
@@ -123,6 +128,134 @@ public class FiltersService : FiltersBase
+                runtimeStream,
+                clientStream,
+                context,
+                _ => _.RegistrationRequest,
+                (serverMessage, registrationResponse) => serverMessage.RegistrationResponse = registrationResponse,
+                (serverMessage, request) => serverMessage.FilterRequest = request,
+                _ => _.FilterResult,
+                _ => _.CallContext,
+                (request, context) => request.CallContext = context,
+                _ => _.CallContext);
+
+            if (!await dispatcher.ReceiveArguments(context.CancellationToken).ConfigureAwait(false))
+            {
+                const string message = "Public Filters connection arguments were not received";
+                _logger.Warning(message);
+                var failure = new Failure(FiltersFailures.NoFilterRegistrationReceived, message);
+                await PublicWriteFailedRegistrationResponse(dispatcher, failure, context.CancellationToken).ConfigureAwait(false);
+                return;
+            }
+
+            var arguments = dispatcher.Arguments;
+            var filterId = arguments.FilterId.To<StreamId>();
+            if (filterId.IsNonWriteable)
+            {
+                _logger.Warning("Received public filter registration request with Filter Id: '{filterId}' which is an invalid stream id", filterId);
+                var failure = new Failure(FiltersFailures.CannotRegisterFilterOnNonWriteableStream, $"Received public filter registration request with Filter Id: '{filterId}' which is an invalid stream id");
+                await PublicWriteFailedRegistrationResponse(dispatcher, failure, context.CancellationToken).ConfigureAwait(false);
+                return;
+            }
+
+            var streamId = StreamId.AllStreamId;
+
+            var registrationResults = await PublicRegisterStreamProcessorsForAllTenants(
+                StreamId.AllStreamId,
+                filterId,
+                dispatcher,
+                context.CancellationToken).ConfigureAwait(false);
+            try
+            {
+                if (!PublicTryStartStreamProcessors(registrationResults, out var failure))
+                {
+                    _logger.Warning(failure.Reason);
+                    await PublicWriteFailedRegistrationResponse(dispatcher, failure, context.CancellationToken).ConfigureAwait(false);
+                    return;
+                }
+
+                await dispatcher.Accept(new FilterRegistrationResponse(), context.CancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                registrationResults
+                    .SelectMany(tenantAndResult => new[] { tenantAndResult.Item2.FilterStreamProcessor })
+                    .ForEach(_ => _.Dispose());
+            }
+        }
+
         bool TryStartStreamProcessors(IEnumerable<(TenantId, FilterRegistrationResult<RemoteFilterDefinition>)> registrationResults, out Failure failure)
```

Comment:
I'm not very fond of the `.Item1` etc. that end up in the code when unnamed tuples are used. If we replace this with something like this, we can refer to the items as normal properties:
```suggestion
        bool TryStartStreamProcessors(IEnumerable<(TenantId Tenant, FilterRegistrationResult<RemoteFilterDefinition> Result)> registrationResults, out Failure failure)
```